gpcc-mocks/src/main/java/uk/nhs/adaptors/gpccmocks/gpc/GpcController.java

```diff
@@ -0,0 +1,208 @@
+import static uk.nhs.adaptors.gpccmocks.common.OperationOutcomes.patientNotFound;
+import static uk.nhs.adaptors.gpccmocks.common.OperationOutcomes.referenceNotFound;
+
+import java.util.List;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestHeader;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.jayway.jsonpath.JsonPath;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import uk.nhs.adaptors.gpccmocks.common.TemplateUtils;
+
+@RestController
+@Slf4j
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class GpcController {
+    @PostMapping(value = "/{odsCode}/STU3/1/gpconnect/structured/fhir/Patient/$gpc.getstructuredrecord")
+    @ResponseStatus(value = ACCEPTED)
+    public ResponseEntity<String> accessStructuredRecord(
+        HttpServletRequest request,
+        @PathVariable String odsCode,
+        @RequestHeader(name = "Ssp-TraceID") String sspTraceId,
+        @RequestHeader(name = "Ssp-From") String sspFrom,
+        @RequestHeader(name = "Ssp-To") String sspTo,
+        @RequestHeader(name = "Ssp-InteractionID") String sspInteractionId,
+        @RequestHeader(name = HttpHeaders.AUTHORIZATION) String authorization,
+        @RequestBody String requestBody) {
+
+        var host = getHostAndPortFromRequest(request);
+
+        log.debug("Request for 'Access Structured Record'. " +
+                "odsCode={} Ssp-TraceID={} Ssp-From={} Ssp-To={} Ssp-InteractionID={} Host/X-Forwarded-Host: {}",
+            odsCode, sspTraceId, sspFrom, sspTo, sspInteractionId, host);
+
+        log.debug("Request body for 'Access Structured Record'\n{}", requestBody);
+
+        if (!isUuid(sspTraceId)) {
+            return badRequest("Ssp-TraceID header must be a UUID");
+        }
+
+        var result = (List<?>) JsonPath.read(requestBody, "$.parameter[:1].valueIdentifier.value");
+        if (result.isEmpty()) {
+            return badRequest("Request body does not include patient identifier");
+        }
+        var nhsNumber = (String) result.get(0);
+
+        GpcModel.GpcModelBuilder gpcModelBuilder = GpcModel.builder()
+            .baseUrl("http://" + host)
+            .odsCode(odsCode);
+
+        switch (nhsNumber) {
+            case "9690937294":
+            case "9690937286":
+                gpcModelBuilder.nhsNumber(nhsNumber);
```

Comment:
In response patient could keeps relation between nhs number and id. Would it be a matter of using the partials/patient in the structured record?

Reply:
The GPCC and GP2GP adaptors don't use the patient's logical id from the ASR for anything. I don't think this has an impact on any further development work. Could you create a backlog ticket for the improvement?

Reply:
ok

Reply:
https://gpitbjss.atlassian.net/browse/NIAD-1566